packages/flutter_tools/test/integration.shard/expression_evaluation_web_test.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+
+import 'package:vm_service/vm_service.dart';
+
+import '../src/common.dart';
+import 'test_data/basic_project.dart';
+import 'test_data/tests_project.dart';
+import 'test_driver.dart';
+import 'test_utils.dart';
+
+void batch1() {
+  final BasicProject _project = BasicProject();
+  Directory tempDir;
+  FlutterRunTestDriver _flutter;
+
+  Future<void> initProject() async {
+    tempDir = createResolvedTempDirectorySync('run_expression_eval_test.');
+    await _project.setUpIn(tempDir);
+    _flutter = FlutterRunTestDriver(tempDir);
+  }
+
+  Future<void> cleanProject() async {
+    await _flutter.stop();
+    tryToDelete(tempDir);
+  }
+
+  Future<void> breakInBuildMethod(FlutterTestDriver flutter) async {
+    await _flutter.breakAt(
+      _project.buildMethodBreakpointUri,
+      _project.buildMethodBreakpointLine,
+    );
+  }
+
+  Future<void> breakInTopLevelFunction(FlutterTestDriver flutter) async {
+    await _flutter.breakAt(
+      _project.topLevelFunctionBreakpointUri,
+      _project.topLevelFunctionBreakpointLine,
+    );
+  }
+
+  test('flutter run expression evaluation - can evaluate trivial expressions in top level function', () async {
+    await initProject();
+    await _flutter.run(withDebugger: true, chrome: true);
+    await breakInTopLevelFunction(_flutter);
+    await evaluateTrivialExpressions(_flutter);
+    await cleanProject();
+  });
+
+  test('flutter run expression evaluation - can evaluate trivial expressions in build method', () async {
+    await initProject();
+    await _flutter.run(withDebugger: true, chrome: true);
+    await breakInBuildMethod(_flutter);
+    await evaluateTrivialExpressions(_flutter);
+    await cleanProject();
+  });
+
+  test('flutter run expression evaluation - can evaluate complex expressions in top level function', () async {
+    await initProject();
+    await _flutter.run(withDebugger: true, chrome: true);
+    await breakInTopLevelFunction(_flutter);
+    await evaluateComplexExpressions(_flutter);
+    await cleanProject();
+  });
+
+  test('flutter run expression evaluation - can evaluate complex expressions in build method', () async {
+    await initProject();
+    await _flutter.run(withDebugger: true, chrome: true);
+    await breakInBuildMethod(_flutter);
+    await evaluateComplexExpressions(_flutter);
+    await cleanProject();
+  });
```

Comment:
All of these need to be skipped if the platform is not Linux, since only those shards have chrome installed

Reply:
got it, thanks!